### PR TITLE
1031 add action to check PR's description format

### LINKED ISSUE #1031: Add commit check for "Fixes #issue" in comment
**What Needs to be Done?**

If a PR is created, users should normally annotate that PR as it fixes an issue so the issue automatically closes when the PR is merged. However, it's easy to forgot to add that to the comment.

Write a PR hook (like the issue style checks) that greps for that in the issue comment so we can inform the user if they forgot.

You can probably model it after the commit check hook, that I've developed in this repo: https://github.com/DARMA-tasking/check-commit-format

### PR DESCRIPTION
**Pull request by jstrzebonski**
Fixes #1031

Most of the work is done in DARMA-tasking/check-pr-fixes-issue#1

The action is run every time the PR title or description (PR's first comment) is edited, and after opening and reopening the PR. If you want to play with it by yourself feel free to check this PR -> jstrzebonski/action_test#9 to not litter this PR anymore.

### CODE REVIEW

File: .github/workflows/check-pr-comment-fixes-issue.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+on:
+  pull_request:
+    types: [opened, edited, reopened]
+
+jobs:
+  check:
+    name: Check PR description
+    runs-on: ubuntu-latest
+    steps:
+      - uses: DARMA-tasking/check-pr-fixes-issue@1031-commit-check-for-fixes-issue
```
Comment (jstrzebonski) [inline .github/workflows/check-pr-comment-fixes-issue.yml:12]: Will be changed to `@master` after merging DARMA-tasking/check-pr-fixes-issue#1

Comment (lifflander) [reply .github/workflows/check-pr-comment-fixes-issue.yml:12]: Sounds good

**Comment by jstrzebonski** [comment]
@lifflander Should this check be `Required`? If so you'll have to change repo settings (I don't have rights to do it :'- ( ). Plus I'll update `wiki/Pull-Requests` paragraph.

**Comment by lifflander** [comment]
> @lifflander Should this check be `Required`? If so you'll have to change repo settings (I don't have rights to do it :'- ( ). Plus I'll update `wiki/Pull-Requests` paragraph.

Yes, I will make it required!

**Comment by lifflander** [comment]
Done, it should be required now.

File: .github/workflows/check-pr-comment-fixes-issue.yml (hunk 1 above)
Comment (lifflander) [reply .github/workflows/check-pr-comment-fixes-issue.yml:12]: It's merged

**Review by lifflander — APPROVED** [review]
This is approved once you switch it to master!

**Comment by jstrzebonski** [comment]
@lifflander Done.

**Comment by lifflander** [comment]
> @lifflander Done.
> 
> Also I've just updated https://github.com/DARMA-tasking/vt/wiki/Pull-Requests with short description of `check-pr-fixes-issue` in 3rd point- `All required checks must pass`

Thanks for updating it!

**Pull request merged**